edweiss-app/components/memento/ModalDisplay.tsx

```diff
@@ -74,46 +73,49 @@ export const CardModalDisplay: ReactComponent<{
                 <TView justifyContent='center' alignItems='center' mb='sm'>
                     <TText bold size='lg' mb='sm'>Card details</TText>
 
-                    <FancyButton
+                    <TTouchableOpacity
                         testID='edit-card'
-                        loading={isLoading}
-                        backgroundColor='transparent'
                         style={{ position: 'absolute', alignSelf: 'flex-end' }}
-                        icon='pencil'
                         onPress={() => {
                             modalRef.current?.dismiss();
-                            router.push({ pathname: `/deck/${id}/card/edition` as any, params: { deckId: id, prev_question: card?.question, prev_answer: card?.answer, cardIndex: absolute_index } })
+                            router.push({ pathname: `/deck/${id}/card/` as any, params: { deckId: id, mode: "Edit", prev_question: card?.question, prev_answer: card?.answer, cardIndex: absolute_index } })
```

Comment:
Same as before, `utils/routeParameters.ts`'s `pushWithParameters(Signature)` might help type safe this sensitive operation. I dealt with these issues in my own app and trust me it's not fun :+1:

Reply:
I have a question: what is ApplicationRouteSignature?

Reply:
I used it in the forum's post route, you can take a look at it.
But in short, `ApplicationRouteSignature` is... well the signature of your route, meaning that it contains all the information necessary to "go there". Mainly, you provide the `path` route, and as a generic value, you provide all the parameters/arguments that you will use inside your route, as such :
```typescript
export const PostRouteSignature: ApplicationRouteSignature<{
    courseId: CourseID,
    postId: Forum.PostID,
    prefetchedPost?: Document<Forum.Post>
}> = {
    path: "/courses/[id]/forum/[postId]"
}
```
You define this right above your route.
Then, to go to this route from another route, you simply use the `pushWithParameters` function as such :
```typescript
pushWithParameters(PostRouteSignature, {
      courseId,
      postId: post.id,
      prefetchedPost: post
})
```
Then, to retrieve all passed parameters inside your route, you use `useRouteParameters` as such :
```typescript
const { courseId, postId, prefetchedPost } = useRouteParameters(PostRouteSignature);
```
This effectively gives you full type safety (if you forget an argument or if it's the wrong type it doesn't compile) and automatic parsing for all argument types (string, number, arrays, objects etc.).

Good luck!

Reply:
Thank you so much for the brief explanation.